From f50c00a545048e2cc509619940f11d12341c532e Mon Sep 17 00:00:00 2001
From: Cristian Sovaiala <cristian.sovaiala@freescale.com>
Date: Fri, 19 Apr 2013 20:56:28 +0000
Subject: [PATCH 154/547] fmd: bypass erratum dTSEC-A003 to allow pause frames
 disable

Bypass Erratum dTSEC-A003 to allow pauseTime=0 to be set
in registry therefore allowing to disable the Pause Frames
on TX side.

dTSEC-A003: Back-to-back pause requests may cause 1588 TX timestamps
	    out of sync

Signed-off-by: Cristian Bercaru <cristian.bercaru@freescale.com>
Signed-off-by: Cristian Sovaiala <cristian.sovaiala@freescale.com>
Change-Id: Ibbc28d2f345c92da8682e8c59dca2e5a37f9be9a
Reviewed-on: http://git.am.freescale.net:8181/1631
Reviewed-by: Radulescu Ruxandra Ioana-B05472 <ruxandra.radulescu@freescale.com>
Reviewed-by: Sovaiala Cristian-Constantin-B39531 <Cristian.Sovaiala@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch taken from QorIQ-SDK-V1.4-SOURCE-20130625-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/net/ethernet/freescale/fman/Peripherals/FM/MAC/dtsec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/fman/Peripherals/FM/MAC/dtsec.c b/drivers/net/ethernet/freescale/fman/Peripherals/FM/MAC/dtsec.c
index e4cb509..6839fcc 100644
--- a/drivers/net/ethernet/freescale/fman/Peripherals/FM/MAC/dtsec.c
+++ b/drivers/net/ethernet/freescale/fman/Peripherals/FM/MAC/dtsec.c
@@ -629,7 +629,7 @@ static t_Error DtsecSetTxPauseFrames(t_Handle h_Dtsec,
 
 #ifdef FM_BAD_TX_TS_IN_B_2_B_ERRATA_DTSEC_A003
     if (p_Dtsec->fmMacControllerDriver.fmRevInfo.majorRev == 2)
-        if (pauseTime <= 320)
+        if (0 < pauseTime && pauseTime <= 320)
             RETURN_ERROR(MINOR, E_INVALID_VALUE,
                      ("This pause-time value of %d is illegal due to errata dTSEC-A003!"
                       " value should be greater than 320."));
-- 
1.8.4.93.g57e4c17

